x-pack/plugins/cloud_security_posture/public/pages/configurations/configurations.tsx

```diff
@@ -15,16 +15,17 @@ import { useLatestFindingsDataView } from '../../common/api/use_latest_findings_
 import { cloudPosturePages, findingsNavigation } from '../../common/navigation/constants';
 import { FindingsByResourceContainer } from './latest_findings_by_resource/findings_by_resource_container';
 import { LatestFindingsContainer } from './latest_findings/latest_findings_container';
+import { getCpmStatus } from '../../common/utils/get_cpm_status';
 
 export const Configurations = () => {
   const location = useLocation();
   const dataViewQuery = useLatestFindingsDataView();
-  const getSetupStatus = useCspSetupStatusApi();
-  const hasFindings =
-    getSetupStatus.data?.indicesDetails[0].status === 'not-empty' ||
-    getSetupStatus.data?.kspm.status === 'indexed' ||
-    getSetupStatus.data?.cspm.status === 'indexed';
-  if (!hasFindings) return <NoFindingsStates posturetype={'cspm'} />;
+  const { data: getSetupStatus } = useCspSetupStatusApi();
+  const status = getCpmStatus(getSetupStatus);
+
+  const noFindingsState = getSetupStatus?.cspm.status !== 'not-installed' ? 'cspm' : 'kspm';
```

Comment:
You can use destruct status and also use `isCspmInstalled` for the no findings state, also rename it to a better name, ex: `noFindingsForPostureType`

```js
 const {hasFindings, isCspmInstalled} = getCpmStatus(getSetupStatus);

 const noFindingsForPostureType = isCspmInstalled ? 'cspm' : 'kspm';

 if (!hasFindings) return <NoFindingsStates posturetype={noFindingsState} />;
```